gensim/models/word2vec.py

```diff
@@ -1229,6 +1209,23 @@ def prepare_vocab(self, hs, negative, wv, update=False, keep_raw_vocab=False, tr
         sample = sample or self.sample
         drop_total = drop_unique = 0
 
+        # if max_vocab is specified instead of min_count
+        # pick a min_count which satisfies max_vocab as well as possible
+        if self.max_vocab is not None:
+            import operator
+
+            sorted_vocab = sorted(self.raw_vocab.items(), key=operator.itemgetter(1), reverse=True)
+            curr_count = 0
+            calc_min_count = 0
+
+            for item in sorted_vocab:
+                curr_count += item[1]
```

Comment:
Each word only counts as 1 in final-vocabulary size, so its actual occurrence count shouldn't be part of any tallying. (If `max_vocab=10`, you just need to throw out all words with the same or fewer occurrences as the 11th word, `sorted_vocab[10]`.)

Reply:
Oh!
I took `max_vocab` to mean maximum number of words
Whereas, it is maximum number of unique words.

I should've realised this sooner!

I thought it would make sense for the user to choose a maximum number of words they'd like. (Would that be good idea as another parameter/option for the user?)